analytics_data_api/utils.py

```diff
@@ -31,3 +31,38 @@ def set_user_auth_token(user, key):
     Token.objects.create(user=user, key=key)
 
     print "Set API key for user %s to %s" % (user, key)
+
+
+def matches(answer_one, answer_two):
+    """ Check whether two answers match sufficiently for consolidation. """
+    return (
+        answer_one.question_text == answer_two.question_text and
+        answer_one.answer_value_text == answer_two.answer_value_text and
+        answer_one.answer_value_numeric == answer_two.answer_value_numeric and
+        answer_one.problem_display_name == answer_two.problem_display_name and
+        answer_one.correct == answer_two.correct and
+        answer_one.value_id == answer_two.value_id
+    )
+
+
+def consolidate_answers(problem):
+    """ Attempt to consolidate erroneously randomized answers. """
+    for answer in problem:
+        answer.consolidated_variant = False
+
+    for i, answer_one in enumerate(problem):
+        j = i + 1
+
+        while j < len(problem):
+            answer_two = problem[j]
+
+            if matches(answer_one, answer_two):
+                answer_one.count += answer_two.count
+                answer_one.variant = None
+                answer_one.consolidated_variant = True
+
+                problem.remove(answer_two)
+            else:
+                j += 1
```

Comment:
What about the case where there's one row that won't consolidate?  e.g.
```
part_id | variant | choice_id | answer text
  a     |  1      |  choice_1 | "foo"
  a     |  2      |  choice_1 | "foo"
  a     |  3      |  choice_1 | "bar"
  a     |  4      |  choice_1 | "baz"
 ```
In this case, I think we ought to be conservative and _not_ consolidate variants 1 and 2, since it's quite possible that the seed is taking effect but just mapping 1 and 2 to the same answer text value.  But this logic would consolidate those 2 cases.

I was thinking that consolidation would only happen if _all_ the rows of a particular `value_id` matched